testlib/src/test/java/com/diffplug/spotless/generic/LicenseHeaderStepTest.java

```diff
@@ -46,6 +54,26 @@ public void fromFile() throws Throwable {
 		assertOnResources(step, KEY_FILE_NOTAPPLIED, KEY_FILE_APPLIED);
 	}
 
+	@Test
+	public void should_apply_license_containing_YEAR_variable() throws Throwable {
+		FormatterStep step = LicenseHeaderStep.createFromFile(createTestFile(KEY_LICENSE_WITH_YEAR), StandardCharsets.UTF_8, LICENSE_HEADER_DELIMITER);
+
+		StepHarness.forStep(step)
+				.test(getTestResource(KEY_FILE_WITHOUT_LICENSE), getFileContentWithYEAR(currentYear()))
+				.testUnaffected(getFileContentWithYEAR(currentYear()))
```

Comment:
From a reproducibility standpoint, using the `currentYear()` instead of a set year like "2003" seems a bit bad to me. Because of this and how there is already a test for "2003", it's not clear to me that testing against `currentYear()` adds any value.

Reply:
Since this token is not something configurable, Spotless will always replace the license with `YearMonth.now()`, so in test its difficult to put here a fixed value.
I see your point on test reproducibility.
If we want to do that, we would need to have more configuration on the license header step:
* token name (optional) -> default `$YEAR`
* closure with the replacement for the token -> default `YearMonth.now.().getYear()`
* the regex to check if the current content of this token is ok (optional) -> default `[0-9]{4}(-[0-9]{4})?`
with that, we could set the configuration in test to fixed values instead of a `moving` one like `YearMonth.now.().getYear()`